internal/authorization/auth_model.go

```diff
@@ -34,5 +40,33 @@ var AuthModel = func() openfga.AuthorizationModel {
 	if err != nil {
 		panic(fmt.Errorf("failed to transform due to %w", err))
 	}
-	return jsonAuthModel
-}()
+
+	return &jsonAuthModel
+}
+
+func (a *AuthorizationModelProvider) prepareModel() *openfga.AuthorizationModel {
+	var model string
+	switch a.apiVersion {
+	case "v2":
+		model = v2Schema
+	default:
+		model = v0Schema
+	}
+
+	return readAuthzModelFromDSLString(model)
+}
```

Comment:
if we have 2 schemas how do we deal with migrations?are the OpenFGA struct methods ready for dealing with separate models?
what is the new schema and how does it change from v0?